### PR TITLE
Use CQL2, clarify data types and request scheams, etc.

### LINKED ISSUE #116: Order request paylod uncertainties
Should the order request properties be returned in the order list?
Should the order list properties be returned in the post request?

### LINKED ISSUE #118: Contraints/Parameters: One Schema or schemas per property?
Is it one big JSON Schema for constraints or multiple json schemas per constraint?

Tendency in discussions seemed to be one schema.

### LINKED ISSUE #119: Order request as JSON Schema not good enough?
How do we formulate queries for orders? JSON Schema? CQL2? ...? JSON Schema is not really meant for queryying data.

### LINKED ISSUE #120: Define Collections
e.g. the OrderCollection and ProductCollection are undefined, but mentioned in the spec

### LINKED ISSUE #132: Update: opportunities may *contain* the Order body in Links (was: Proposal: Define a data type that is 1. input to opportunities 2. returned by opportunities in a list 3. the input to the orders endpoint)
The datatype (Should we call it Search? or Order-like? calling it `O` for now) would be similar to the current Order entity, ie would need to contain `product_id`, `datetime`, and `geometry`, with some open discussion on how to represent additional constraints. 

In any case, it serves similar purposes in each of the three scenarios:
1. `O` describes the search of opportunities for a given product
2. `O` describes the found opportunities for a given product, this could just be 
- the provided `O` (to confirm feasibility) or
- a list of `O` derived from the input `O` to narrow down e.g. datetime and geometry for specific imaging windows
3. Once the user decides for one of the `O`s and wants to make the order, they can simply push the received `O` to the orders endpoint. We identified that an order, in the end, is still a search in the sense that you send constraints even though they may or may not be more narrow than the original search)

Users might also decide to bypass the opportunities endpoint and send their `O` directly to the orders endpoint. Whether this results in a useful order is up to the implementation.

### PR DESCRIPTION
**Pull request by m-mohr**
Update the constraints to use CQL2 as discussed + cleanup properties as discussed in the Wed morning session. Fixes #119

Also adds docs for the Order and Opportunity Collections. Fixes #120

Update README with accepts column. Fixes  #116

Clairfy JSON Schema requirement for constraints. Fixes #118 and #119

Clarify Accept & Return values from endpoints. FIxes #132

Remove GET /opportunities because it's undefined. Partially solves #98

### CODE REVIEW

**Review by CloudNiner — APPROVED** [review]

**Comment by nic-planet** [comment]
I just wrote up something related here https://github.com/Element84/stat-api-spec/issues/132 would be great if we could align on that

File: README.md [hunk 1]
```diff
@@ -43,15 +43,15 @@ The following table describes the service resources available in a STAT API impl
...
+| `GET /orders`               | Core          | -                                                            | [Orders Collection](./order/README.md#order-collection)      |                                                              |
+| `POST /orders`              | Core          | [Order Request](./order/README.md#order-request)             | [Order Object](./order/README.md#order-pobject)              | Order a capture with a particular set of constraints         |
+| `GET /opportunities`        | Opportunities | -                                                            | [Opportunities Collection](./opportunity/README.md#opportunities-collection) |                                                              |
+| `POST /opportunities`       | Opportunities | [Opportunity Request](./opportunity/README.md#opportunity-request) | [Opportunity Object](./opportunity/README.md#opportunity-object) | Explore the opportunities available for a particular set of constraints |
```
[the remark this answers is not in the document]
Comment (m-mohr) [reply README.md]: Updated accordingly.

**Comment by m-mohr** [comment]
> I just wrote up something related here #132 would be great if we could align on that

Sure. Maybe merge this and then update afterwards?

File: README.md [hunk 2]
```diff
@@ -43,15 +43,14 @@ The following table describes the service resources available in a STAT API impl
...
+| `GET /`                     | Core          | -                                                            | [Landing Page](#landing-page)                                |                                                              |
+| `GET /products`             | Core          | -                                                            | [Products Collection](./product/README.md)                   | Figure out which constraints are available for which `product_id` |
+| `GET /products/{productId}` | Core          | -                                                            | [Product](./product/README.md)                               |                                                              |
+| `GET /orders`               | Core          | -                                                            | [Orders Collection](./order/README.md#order-collection)      |                                                              |
+| `POST /orders`              | Core          | [Order Request](./order/README.md#order-request)             | [Order Object](./order/README.md#order-pobject)              | Order a capture with a particular set of constraints         |
```
Comment (nic-planet) [inline README.md:52]: I'm not sure that we need a distinction between an OrderRequest and an Order.

Comment (m-mohr) [reply README.md:52]: That's how it is documented now and is not coming in new in this PR.

Comment (nic-planet) [reply README.md:52]: oh very true. misread the merge

**Review by nic-planet — APPROVED** [review]
Approve, as discussed let's review separately if input and output of opportunities and input to orders could be the same separately

**Review by DylanBartels — APPROVED** [review]

File: product/README.md [hunk 3]
```diff
@@ -59,35 +58,28 @@ For a full discussion of the situations where relative and absolute links are re
...
+to align with the JSON Schemas provided by STAC. Empty schemas are not allowed.
 
-1. A set of all distinct values in an array: The set of values must contain at least one element and it is strongly recommended to list all values.
-2. A Range in a [Range Object](#range-object): Statistics by default only specify the range (minimum and maximum values),
-   but can optionally be accompanied by additional statistical values.
-   The range specified by the `minimum` and `maximum` properties can specify the potential range of values,
-   but it is recommended to be as precise as possible.
-3. Extensible JSON Schema definitions for fine-grained information, see the [JSON Schema Object](#json-schema-object)
-   section for more.
+For an introduction to JSON Schema, see "[Learn JSON Schema](https://json-schema.org/learn/)".
```
Comment (jp-u) [inline product/README.md:71]: I'm getting a 404 on this link. Is there a working URL?

Comment (m-mohr) [reply product/README.md:71]: Probably an outdated URI. A PR would be appreciated.

**Pull request merged**